src/main/java/org/mdbenefits/app/submission/actions/HandleApplicationSigned.java

```diff
@@ -19,18 +20,29 @@ public class HandleApplicationSigned implements Action {
 
     private final SubmissionRepositoryService submissionRepositoryService;
 
+    private final JdbcTemplate jdbcTemplate;
+
     public HandleApplicationSigned(MessageSource messageSource, MailgunEmailClient mailgunEmailClient,
-            SubmissionRepositoryService submissionRepositoryService) {
+            SubmissionRepositoryService submissionRepositoryService, JdbcTemplate jdbcTemplate) {
         this.messageSource = messageSource;
         this.mailgunEmailClient = mailgunEmailClient;
         this.submissionRepositoryService = submissionRepositoryService;
+        this.jdbcTemplate = jdbcTemplate;
     }
 
     @Override
     public void run(Submission submission) {
+        assignConfirmationNumber(submission);
         sendEmailToApplicant(submission);
     }
 
+    private void assignConfirmationNumber(Submission submission) {
+        String confirmationNumber = jdbcTemplate.queryForObject("select 'M-' || nextval('confirmation_sequence')", String.class);
+        submission.getInputData().put("confirmationNumber", confirmationNumber);
+        log.info("Using confirmationNumber: {}", confirmationNumber);
```

Comment:
I like what you did with this. Can you add the submission ID to this log message? I could see that being useful during debug (rather than having to look it up separately).

Reply:
I didn't add it as it is in the MDC.

Reply:
Oh, I forgot about that.  Never mind then.  :)